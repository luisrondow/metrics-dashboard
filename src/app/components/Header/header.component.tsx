import { StyledHeader } from './header.styles';

const Header = () => {
  return (
    <StyledHeader>
      <h1>Metrics Dashboard</h1>
    </StyledHeader>
  );
};

export default Header;
