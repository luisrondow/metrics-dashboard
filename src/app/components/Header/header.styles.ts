import styled from "styled-components";

export const StyledHeader = styled.header`
  background-color: #282c34;
  color: white;
  padding: var(--space-md);
  display: flex;
  justify-content: center;
  align-items: center;
`;