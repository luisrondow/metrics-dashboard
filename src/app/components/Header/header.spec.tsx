import { render, screen } from '@testing-library/react';

import Header from './header.component';

describe('Header Component', () => {
  // add a snapshot test
  it('matches snapshot', () => {
    const { asFragment } = render(<Header />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the header with correct text', () => {
    render(<Header />);
    const headerElement = screen.getByText('Metrics Dashboard');
    expect(headerElement).toBeInTheDocument();
  });

  it('renders the header element', () => {
    render(<Header />);
    const headerElement = screen.getByRole('banner');
    expect(headerElement).toBeInTheDocument();
  });

  it('renders the h1 element inside the header', () => {
    render(<Header />);
    const h1Element = screen.getByRole('heading', { level: 1 });
    expect(h1Element).toBeInTheDocument();
    expect(h1Element).toHaveTextContent('Metrics Dashboard');
  });
});
