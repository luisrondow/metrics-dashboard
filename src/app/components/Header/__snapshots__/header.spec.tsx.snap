// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="sc-5dbe63cf-0 giLltp"
  >
    <h1>
      Metrics Dashboard
    </h1>
  </header>
</DocumentFragment>
`;
